crates/re_space_view_spatial/src/scene/parts/entity_iterator.rs

```diff
@@ -0,0 +1,70 @@
+use re_log_types::{Component, EntityPath};
+use re_query::{query_primary_with_history, EntityView, QueryError};
+use re_renderer::DepthOffset;
+use re_viewer_context::{ArchetypeDefinition, SceneQuery, SpaceViewHighlights, ViewerContext};
+
+use crate::scene::contexts::{SpatialSceneContext, SpatialSceneEntityContext};
+
+/// Iterates through all entity views for a given archetype.
```

Comment:
This snippet is worth reviewing with @teh-cmc as a separate standalone discussion outside of this PR. It seems like we will want to do something very similar if not identical to this for every SpaceView, so figuring out how to generalize the spatial-specific details and then otherwise move this into the Facade would be helpful.

Reply:
btw. worth noting that I just unified existing code there, this is not really new and part of the framework as such but I didn't want to leave it duplicated

Reply:
Yeah; just noting that this feels like one of the heavy facade/archetype'ish patterns.